File: sdk/src/types/file.ts
/** Clean formatted shape (`id`, not `_id`) — unlike Thread/Skill/Agent/Knowledge. */
export interface PersonaFile {
  id: string;
  originalName: string;
  mimeType: string;
  size: number;
  agentId: string | null;
  threadId: string | null;
  createdAt: string;
}

export interface UploadFilePayload {
  filename: string;
  /** Node Buffer/Uint8Array, or an already-constructed Blob. */
  content: Uint8Array | Blob;
  contentType?: string;
  agentId?: string;
  threadId?: string;
}

export interface ListFilesParams {
  page?: number;
  limit?: number;
}

File: sdk-python/src/personaai/types/file.py
"""Mirrors ``sdk/src/types/file.ts``."""

from __future__ import annotations

from typing import BinaryIO, TypedDict


class PersonaFile(TypedDict):
    """Clean formatted shape (``id``, not ``_id``) — unlike Thread/Skill/Agent/Knowledge."""

    id: str
    originalName: str
    mimeType: str
    size: int  # bytes
    agentId: str | None
    threadId: str | None
    createdAt: str


class _UploadFilePayloadRequired(TypedDict):
    filename: str
    content: bytes | BinaryIO


class UploadFilePayload(_UploadFilePayloadRequired, total=False):
    contentType: str  # e.g. 'application/pdf'; required when content type can't be inferred
    agentId: str  # associates this file with an Agent, e.g. for later reference in tool calls
    threadId: str  # associates this file with a specific Thread/conversation


class ListFilesParams(TypedDict, total=False):
    page: int  # default: 1
    limit: int  # default: 20
